### PR TITLE
Swap in base58_check for base64 in most places

### PR DESCRIPTION
**Pull request by psteckler**
Swap in base58_check in most places that base64 was being used. Didn't do it for lite client, can do that if this PR is OK. 

Tagged this PR as WIP to see if CI passes.

Renamed `Base58_check.base58_check` to `Base58_check.encode` and `Base58_check.base58_check_decode_exn` to `Base58_check.decode_exn`.

### CODE REVIEW

File: src/app/cli/src/graphql.ml [hunk 1]
```diff
@@ -45,17 +45,20 @@ module Types = struct
...
 
-    module State_hash = Codable.Make_base64 (State_hash.Stable.V1)
+    module State_hash = Codable.Make_base58_check (State_hash.Stable.V1)
   end
 
   module Id = struct
-    (* The id of a user_command is base64 the seralized version of the user_command *)
+    let version_byte = '\x17'
+
+    (* The id of a user_command is.encode of the seralized version of the user_command *)
```
Comment (bkase) [inline src/app/cli/src/graphql.ml]: this comment got mangled

File: src/app/cli/src/graphql.ml [hunk 2]
```diff
@@ -45,17 +45,20 @@ module Types = struct
 
     let amount amount = uint64 @@ Currency.Amount.to_uint64 amount
 
-    module State_hash = Codable.Make_base64 (State_hash.Stable.V1)
+    module State_hash = Codable.Make_base58_check (State_hash.Stable.V1)
   end
 
   module Id = struct
-    (* The id of a user_command is base64 the seralized version of the user_command *)
+    let version_byte = '\x17'
```
Comment (bkase) [inline src/app/cli/src/graphql.ml]: How are all these `version_byte` values chosen? Should we gather them in one variant somewhere so we can easily tell that they are all distinct?

Comment (psteckler) [reply src/app/cli/src/graphql.ml]: Chosen randomly. We can create a file of them in `base58_check`, if you'd like.

File: src/app/cli/src/graphql.ml (hunk 1 above)
Comment (psteckler) [reply src/app/cli/src/graphql.ml]: sed fail.

File: src/app/cli/src/graphql.ml (hunk 2 above)
Comment (bkase) [reply src/app/cli/src/graphql.ml]: Yes that seems like a good idea

Comment (psteckler) [reply src/app/cli/src/graphql.ml]: Done.

File: src/app/cli/src/graphql.ml (hunk 1 above)
Comment (psteckler) [reply src/app/cli/src/graphql.ml]: Fixed.

**Comment by psteckler** [comment]
> Should we just start at 0x00 and increment?

We can do that, but we may want particular values in order to get base58 strings that begin with certain characters to make them recognizable as a given type. I think Bitcoin does that.

Maybe it's enough to put them in numeric order?

**Pull request merged**